src/js/components/tasks/retrieveDataset/data.tsx

```diff
@@ -0,0 +1,518 @@
+    change,
+    checked,
+  }: {
+    groupName: string;
+    change: string;
+    checked: boolean;
+  }) => void;
+  className?: string;
+}) => {
+  const { t } = useTranslation();
+  const [expandedPanels, setExpandedPanels] = useState<BooleanObject>({});
+  const [filteredSchema, setFilteredSchema] = useState(schema);
+  const [search, setSearch] = useState<string>('');
+
+  let truncatedMsg = null;
+  let sortedSchema = sortSchema(filteredSchema);
+  const noSchema = isEmpty(filteredSchema);
+  if (sortedSchema.length > SCHEMA_SIZE_LIMIT) {
+    sortedSchema = sortedSchema.slice(0, SCHEMA_SIZE_LIMIT);
+    truncatedMsg = t(
+      'Only displaying the first {{limit}} objects. Enter a term above to search the full list.',
+      { limit: SCHEMA_SIZE_LIMIT },
+    );
+  }
+
+  const handlePanelToggle = (groupName: string) => {
+    setExpandedPanels({
+      ...expandedPanels,
+      [groupName]: !expandedPanels[groupName],
+    });
+  };
+
+  // Update schema when search term changes
+  useEffect(() => {
+    setFilteredSchema(
+      search ? filterSchema(schema, search.toLowerCase()) : schema,
+    );
+  }, [schema, search]);
+
+  let emptyHeading =
+    type === 'all' ? t('No data available to retrieve') : t('No data selected');
+  let emptyMsg =
+    type === 'all'
+      ? null
+      : t(
+          'Choose objects or fields from the left panel to add them to this dataset.',
+        );
+  if (search) {
+    emptyHeading = t('No data found');
+    emptyMsg = t(
+      'Change or remove your search term above to view additional data.',
+    );
+  }
+
+  const paddingClasses =
+    'slds-m-left_xx-small slds-p-horizontal_medium slds-p-right_medium';
+
+  return (
+    <div
+      className={classNames('has-checkboxes', 'slds-form', className)}
+      data-form="task-retrieve-changes"
+      {...props}
+    >
+      <ModalCard
+        heading={
+          <div className="slds-m-left_xx-small">
+            <SearchForm truncatedMsg={truncatedMsg} setSearch={setSearch} />
+            <p>{heading}</p>
+          </div>
+        }
+        noBodyPadding
+      >
+        {errors ? (
+          <p
+            className={classNames(
+              'slds-text-color_error',
+              'slds-p-top_x-small',
+              paddingClasses,
+            )}
+          >
+            {errors}
+          </p>
+        ) : null}
+        {noSchema ? (
+          <EmptyIllustration heading={emptyHeading} message={emptyMsg} />
+        ) : (
+          <>
+            <div
+              className="form-grid-three
+                slds-m-left_xx-small
+                slds-p-left_x-large
+                slds-p-right_medium
+                slds-p-vertical_x-small"
+            >
+              <span>
+                <strong>{t('Label')}</strong>
+              </span>
+              <span>
+                <strong>{t('Developer Name')}</strong>
+              </span>
+              <span>
+                <strong>{t('Records')}</strong>
```

Comment:
I know this touches the right border at some screen sizes, but I though that was better than:
a. writing some media queries to give it a negative left margin at some sizes
b. reworking the grid giving less room for the two more verbose columns
c. making the text smaller

Reply:
We occasionally have side-scroll when Field names get long, but I think this is okay as-is:

![Screen Shot 2022-07-26 at 10 59 25 AM](https://user-images.githubusercontent.com/552316/181040583-05191b57-5ccf-4ff2-977c-7385e1a88f58.jpg)